src/main/java/com/squareup/javapoet/CodeWriter.java

```diff
@@ -247,6 +264,19 @@ public CodeWriter emit(CodeBlock codeBlock) throws IOException {
           break;
 
         default:
+          ClassName pendingClassName = potentiallyStaticImported;
+          potentiallyStaticImported = null;
+          if (pendingClassName != null && part.startsWith(".")) {
+              String partWithoutLeadingDot = part.substring(1);
+              String memberName = Util.identifier(partWithoutLeadingDot);
```

Comment:
Good catch. My mind is still using the `$R` syntax, in which these edge cases are not present.